staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -666,6 +745,35 @@ func (p *sharedProcessor) resyncCheckPeriodChanged(resyncCheckPeriod time.Durati
 	}
 }
 
+func (p *sharedProcessor) removeListenerFor(handler ResourceEventHandler) {
+	p.listenersLock.Lock()
+	defer p.listenersLock.Unlock()
+
+	listener := p.removeListenerLockedFor(handler)
+	if p.listenersStarted && listener != nil {
+		close(listener.addCh)
+	}
+}
+
+func (p *sharedProcessor) removeListenerLockedFor(handler ResourceEventHandler) *processorListener {
+	if !reflect.ValueOf(handler).Type().Comparable() {
+		return nil
+	}
+	var listener *processorListener
+	for i, l := range p.listeners {
```

Comment:
Note that you have made a decision to remove all occurrences of the handler, in the event that a handler is added multiple times. That's probably fine (and there's not really a good reason to ever add a handler more than once), but I would explicitly call out this decision in a comment somewhere, in case users are expecting different behavior.

Reply:
I add a note to the method definition and interface comment